model/service.go

```diff
@@ -25,10 +25,15 @@ import (
 type ServiceDiscovery interface {
 	// Services list declarations of all services and their tags
 	Services() []*Service
+
+	// GetService retrieves a service by name if it exists
+	GetService(name, namespace string) (*Service, bool)
```

Comment:
How about making it a bit more generic, such as GetService(name, namespace, tags) ?
Ideally, keeping in line with the ClusterIdentifier in the proto description, we should be using only name, tags as argument. (you would then tuck namespace into name field, there by hiding kube specific namespace concept from the model).

The model is supposed to be independent of the underlying platform. Namespace concept is kubernetes specific. It will break in Mesos/VM cluster or whatever. WDYT?